scripts/cy-wrapper.sh

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
+
+if test -z "$BASH_VERSION"; then
+  echo "Please run this script using bash, not sh or any other shell." >&2
+  exit 1
+fi
+
+export CY_API_URL="${CY_API_URL:-https://http-api.cycloid.io}"
+export CY_BINARIES_PATH="${CY_BINARIES_PATH:-$HOME/.cycloid}"
+
+# Ensure we have Cycloid directory created
+if ! [ -d "$CY_BINARIES_PATH" ]; then
+  mkdir -p $CY_BINARIES_PATH
+fi
+
+# If specified in the commandline use it
+if [[ `echo "$@" | grep -E --  "--api-url(=|\s+)([^\]+)"` ]]; then
+  CY_API_URL=$(echo "$@" | sed -r "s/.*--api-url(=|\s+)([^ ]+).*/\2/")
+fi
+
+# Remove trailing /
+CY_API_URL=${CY_API_URL%/}
+
+# Get Cycloid API version
+export CY_VERSION=$(curl -s "${CY_API_URL}/version" | jq -r .data.version)
+
+if [[ -z "$CY_VERSION" ]]; then
+  echo "Error: Unable to get Cycloid API version on ${CY_API_URL}/version"
+  exit 1
+fi
+
+# Download the binary if not present
+export CY_BINARY="${CY_BINARIES_PATH}/${CY_VERSION}"
+if ! [[ -f "${CY_BINARY}" ]]; then
+  CY_URL="https://github.com/cycloidio/cycloid-cli/releases/download/v${CY_VERSION}/cy"
+  wget -q -O "${CY_BINARY}" "$CY_URL"
+  if [ $? != 0 ]; then
+    rm "${CY_BINARY}"
+    echo "Error: Unable to download Cycloid CLI from github $CY_URL"
+    exit 1
+  fi
+  chmod +x "${CY_BINARY}"
+fi
+
+# Run Cycloid CLI
+exec "${CY_BINARY}" $@
```

Comment:
```suggestion
exec "${CY_BINARY}" "$@"
```
From [GNU Bash manual --> 3.4.2 Special Parameters](https://www.gnu.org/software/bash/manual/html_node/Special-Parameters.html):

> ($@) Expands to the positional parameters, starting from one. **When the expansion occurs within double quotes, each parameter expands to a separate word. That is, "$@" is equivalent to "$1" "$2"** …. If the double-quoted expansion occurs within a word, the expansion of the first parameter is joined with the beginning part of the original word, and the expansion of the last parameter is joined with the last part of the original word. When there are no positional parameters, "$@" and $@ expand to nothing (i.e., they are removed).